src/Components/Components/src/ElementRef.cs

```diff
@@ -45,13 +45,28 @@ static string CreateUniqueId()
                 // fields for ElementRefCaptureId, of which only one would be in use depending
                 // on the platform.
                 var id = Interlocked.Increment(ref _nextIdForWebAssemblyOnly);
-                return id.ToString();
+                return id.ToString(CultureInfo.InvariantCulture);
             }
             else
             {
                 // For remote rendering, it's important not to disclose any cross-user state,
                 // such as the number of IDs that have been assigned.
-                return Guid.NewGuid().ToString("D");
+                return Guid.NewGuid().ToString("D", CultureInfo.InvariantCulture);
+            }
+        }
+
+        internal sealed class ElementRefConverter : JsonConverter<ElementRef>
+        {
+            private static readonly JsonEncodedText IdProperty = JsonEncodedText.Encode("__internalId");
+
+            public override ElementRef Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+                => throw new NotSupportedException();
```

Comment:
We should add support for this while we're here, there's no good reason why this shouldn't roundtrip.